internal/handlers/tigris/msg_distinct.go

```diff
@@ -42,18 +43,28 @@ func (h *Handler) MsgDistinct(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg
 		return nil, err
 	}
 
+	filter := dp.Filter
+
 	qp := tigrisdb.QueryParams{
 		DB:         dp.DB,
 		Collection: dp.Collection,
-		Filter:     dp.Filter,
 	}
 
-	resDocs, err := fetchAndFilterDocs(ctx, &fetchParams{dbPool, &qp, h.DisableFilterPushdown})
+	if !h.DisableFilterPushdown {
+		qp.Filter = filter
```

Comment:
`filter` variable is used only once, right? isn't it possible to do something like `qp.Filter = dp.Filter`? I don't really understand why we need a new variable there.

Reply:
It's used twice: here to setup queryparam filter, and on the line 65 to apply filters after fetching documents. We could remove this variable and get value from `dp.Filter` in both places, although it seemed to me a bit unclear as it's easy to confuse `qp.Filter` with `dp.Filter`.